# Parlament Speaker Recogniction Model
This notebook should predict what speaker said certain things based on the Austrian parlament speaches. 

This small project is based on the Dataset assembled by Mario Zechner (https://marioslab.io/)

In [5]:
# DATA FROM SPEAKER
import pandas as pd

# Either use direct download or load locally https://woswormeileistung.marioslab.io/data/persons.json
#speaker = pd.read_json("https://woswormeileistung.marioslab.io/data/persons.json")
speaker = pd.read_json("persons.json")
print(speaker.shape)
speaker.head()

(681, 6)


id                        name      parties  \
0   3612              Sonja Ablinger        [SPÖ]   
1  14854  Dipl.-Ing. Elke Achleitner   [BZÖ, FPÖ]   
2  83119           Mag. Nikolaus Alm  [LIF, NEOS]   
3  51879   Mag. Hannes Amesbauer, BA        [FPÖ]   
4   2819            Werner Amon, MBA        [ÖVP]   

                                        periods  \
0                             [XX, XXIII, XXIV]   
1                                        [XXII]   
2                                         [XXV]   
3                                 [XXVI, XXVII]   
4  [XIX, XX, XXI, XXII, XXIII, XXIV, XXV, XXVI]   

                                    url  \
0   https://parlament.gv.at/person/3612   
1  https://parlament.gv.at/person/14854   
2  https://parlament.gv.at/person/83119   
3  https://parlament.gv.at/person/51879   
4   https://parlament.gv.at/person/2819   

                                            imageUrl  
0  https://parlament.gv.at/dokument/bild/34898/34...  
1  https://parlament.gv.at/dokument/bild/21013/21...  
2  https://parlament.gv.at/dokument/bild/43808/43...  
3  https://parlament.gv.at/dokument/bild/200696/2...  
4  https://parlament.gv.at/dokument/bild/90356/90...

As one can see there are about 681 Speakers from the austrian parlament, starting from 2002

In [ ]:
# DATA FROM SESSIONS 
# BE AWARA: sessions data has 440 mb, so it is only imported locally 
# https://woswormeileistung.marioslab.io/data/sessions.json

sessions = pd.read_json("sessions.json")

print(df.shape)
df.head()